protocol/communities/persistence.go

```diff
@@ -1632,3 +1632,70 @@ func (p *Persistence) SaveRequestsToJoin(requests []*RequestToJoin) (err error)
 	err = tx.Commit()
 	return err
 }
+
+func (p *Persistence) GetCuratedCommunities() (*CuratedCommunities, error) {
+	rows, err := p.db.Query("SELECT community_id, featured FROM curated_communities")
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	result := &CuratedCommunities{
+		ContractCommunities:         []string{},
+		ContractFeaturedCommunities: []string{},
+	}
+	for rows.Next() {
+		var communityID string
+		var featured bool
+		if err := rows.Scan(&communityID, &featured); err != nil {
+			return nil, err
+		}
+		result.ContractCommunities = append(result.ContractCommunities, communityID)
+		if featured {
+			result.ContractFeaturedCommunities = append(result.ContractFeaturedCommunities, communityID)
+		}
+	}
+
+	return result, nil
+}
+
+func (p *Persistence) SetCuratedCommunities(communities *CuratedCommunities) error {
+	tx, err := p.db.BeginTx(context.Background(), &sql.TxOptions{})
```

Comment:
i wish we can boilerplate transaction someday (not a comment, just sharing)

Reply:
Wdym exactly?

Reply:
we have this code repeated dozens of times. Just curious, if it's possible to boilerplate it 
```
tx, err := p.db.BeginTx(context.Background(), &sql.TxOptions{})

if err != nil {
		return err
	}

defer func() {
		if err == nil {
			err = tx.Commit()
			return
		}
		// don't shadow original error
		_ = tx.Rollback()
	}()
```

Reply:
Ah, would be nice indeed. The most obvious would be to use helper function:

```
defer func() {
    FinalizeTransaction(tx, &err)
}()
```